### ISSUE TITLE
[Feature Request] Support for pressure-sensitive face buttons

### ISSUE THREAD

**Issue by eVenent** [xemu-project/xemu#606]
### Feature Request

The Xbox OG Controller drivers in perspective of pressure sensitive buttons are garbage. XBCD allows to use assign 6 of 8 pressure sensitive buttons to 3 free axis. Mayflash Tigergame allows only 2 of 8.  Recommended Xb2XInput none as translates signals to XInput which is not supporting pressure sensitive buttons. 

I saw that [Cxbx have added recently support for Xbox Controller](https://github.com/Cxbx-Reloaded/Cxbx-Reloaded/pull/2290) (and [SBC](https://github.com/faha223/SteelBattalionController)) via USB passthrough with libusb. Is it possible to add support for pressure sensitive buttons to Xbox Controller via libusb too?

Additionally, there are 2 other controllers, DualShock 2 and DualShock 3 with pressure sensitive support too. I saw that RPCS3 and PCSX2 have added support for DualShock 3's pressure sensitive buttons using [DsHidMini](https://github.com/ViGEm/DsHidMini) driver. Maybe it will be possible to make xemu compatible with PlayStation's pressure sensitive buttons too?

I think that pressure sensitive buttons are something unique, which should not be forgotten and it's worth to preserve support for them.

### Alternatives

Nothing interesting as alternative. Pressure sensitivity buttons are unique feature which is difficult to replace.
- Use existing drivers to implement pressure sensitive buttons (some buttons can be missed).
- Emulate pressure sensitivity - pressed button is reaching sensitivity from 0% to 100% in ~0.5s, so fast tapping will let simulate soft press, and long press will simulate hard press.

### Additional Context

PC gamers will be able to feel difference in games which supports pressure sensitivity comparing them to PC ports. Xbox gamers will be able to play their games like on their old device. It will lead to better immersion and higher satisfaction. Anyway feature can be optional and should be disabled for people who does not enjoy using pressure-sensitive buttons.

**Comment by GXTX** [comment]
> Is it possible to add support for pressure sensitive buttons to Xbox Controller via libusb too?

No need. Pass the physical USB device into xemu, no need for Windows drivers. See QEMUs docs on how to pass physical USB devices to the guest VM.

> Maybe it will be possible to make [...] PlayStation's pressure sensitive buttons too?

xemu uses SDL for it's controller backend. You'd need to suggest such things upstream to them.

**Comment by Spidy123222** [comment]
> > Is it possible to add support for pressure sensitive buttons to Xbox Controller via libusb too?
> 
> 
> 
> No need. Pass the physical USB device into xemu, no need for Windows drivers. See QEMUs docs on how to pass physical USB devices to the guest VM.
> 
> 
> 
> > Maybe it will be possible to make [...] PlayStation's pressure sensitive buttons too?
> 
> 
> 
> xemu uses SDL for it's controller backend. You'd need to suggest such things upstream to them.
> 
> 

I agree with this this should be on SDL side. Also there are gamepads other than original Xbox that did have pressure sensitive buttons like p2 and ps3 dual shock controllers which passthrough wouldn't work with since it's not a actual Xbox controller. I'm no technical person so idk if you could passthrough a ps3 controller. Which was stated in OG post. So this belongs in sdl more of.

**Comment by eVenent** [comment]
So it finally happened for DualShock 3 with this [commit](https://github.com/libsdl-org/SDL/commit/aaee6e242c6309e33ac67c70ed96183beef6a75c) and SDL is supporting pressure-sensitive face buttons. 😁 For now, supports only DS3 and DS2 adapters which convert signals to DS3 (like Ostent ones), it does not have Xbox OG pressure-sensitive buttons yet, but maybe it will change in the future. BUT DS3 controllers for sure will cover a big chunk of the total number of use cases.

**Comment by eVenent** [comment]
I have one interesting update. 

_ReWASD team has introduced [Virtual DS3 controller](https://www.rewasd.com/releases/release-6.3.0)  which can emulate pressure-sensitive buttons on every supported controller. It means that even Xbox One controllers can have pressure-sensitive options depends how we will configure it (I'm usually using other button modifiers or combos like RB+A i giving only 20% pressure, but normal press A is 100% or short press 20%, double press 50% long press 100%). This solution is tested for PS emulators and works great._

They even modified PCSX2's code to support native DS3 driver for pressure-sensitivity - [Native DualShock 3 support in Windows by official Sony driver (sixaxis)](https://github.com/PCSX2/pcsx2/pull/7099). I think it's worth to implement this feature in this way by adding support for official [DS3 driver](https://www.pcgamingwiki.com/wiki/Controller:DualShock_3#Wired_connection_-_official_drivers) and the rest job can be done by rewasd's virtual DS3 controller or the real one.

**Comment by Jobima1st** [comment]
> xemu uses SDL for it's controller backend. You'd need to suggest such things upstream to them.

isnt SDL3 now should support Pressure sensitive buttons since this https://github.com/libsdl-org/SDL/issues/5148 closed as 100% fixed ?
